transports/websocket/src/framed.rs

```diff
@@ -190,10 +190,18 @@ where
                         }
 
                         let ws_key = {
-                            let request = server.receive_request()
-                                .map_err(|e| Error::Handshake(Box::new(e)))
-                                .await?;
-                            request.into_key()
+                            let response = server.receive_request()
+                                .await;
+                            match response {
+                                Ok(request) => request.into_key(),
+                                Err(e) => {
```

Comment:
Here one should probably inspect the error a bit more (cf. [`soketto::handshake::Error`](https://docs.rs/soketto/0.3.1/soketto/handshake/enum.Error.html)). I do not think it makes sense to fall back in case of an I/O error for example. Instead a fallback should maybe only be attempted in case of an [HTTP parsing error](https://docs.rs/soketto/0.3.1/soketto/handshake/enum.Error.html#variant.Http).